Move static VRF variables to namespace

jmythms opened the pull request:
Pull request overview
---------------------
 - Partially fixes #7729 
 - This moves the static variables in the HVACVariableRefrigerantFlow.cc file to the namespace. Multiple condensers might still not work. The HRTime and HRTimer variables are still in the code. 

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

jmythms committed:
Changing static variables to namesake variables

Partial fix to #7729

jmythms committed:
Merge branch 'develop' into Refactor-VRF-static-timing-variables-#7729

jmythms committed:
Merge branch 'develop' into Refactor-VRF-static-timing-variables-#7729

jmythms commented on src/EnergyPlus/HVACVariableRefrigerantFlow.cc:
```diff
@@ -220,6 +220,9 @@ namespace HVACVariableRefrigerantFlow {
     Array1D<Real64> MinDeltaT;              // minimum zone temperature difference from setpoint
     Array1D<Real64> SumCoolingLoads;        // sum of cooling loads
     Array1D<Real64> SumHeatingLoads;        // sum of heating loads
+    Real64 CurrentEndTime;                  // end time of current time step
+    Real64 CurrentEndTimeLast;              // end time of last time step
+    Real64 TimeStepSysLast;                 // system time step on last time step
```
The three variables are not initialized. 

The static variables in the original code were not initialized as well, so I am not sure if this is ok.

mitchute replied:
It's fine without initial values unless it's complaining about them being used before being initialized. That would need to be corrected.

jmythms committed:
Hotfix #7723 revert + Add flag to reset CurrentEndTimeLast when InitVRF is not called

The unit test will work without Hotfix #7723.

Not sure if multiple condensors will work.
The HRTime, HRTimer variables still exist.

jmythms committed:
Merge branch 'develop' into Pull-static-variables-to-namespace

mitchute commented:
@Myoldmopar CI didn't pick this up for some reason. Is there any way to restart it? Or do we just bump it again with another minor change?
